packages/lib/src/data-grid/DataGrid.tsx

```diff
@@ -0,0 +1,329 @@
+          colSpan(args) {
+            return args.type === "ROW" && args.row.isExpandedChildContent
+              ? columns.length + (selectable ? 2 : 1)
+              : undefined;
+          },
+          renderCell({ row }) {
+            if (row.isExpandedChildContent) {
+              // if it is expanded content
+              return row.expandedChildContent || <></>;
+            }
+            // if row has expandable content
+            return (
+              <ActionContainer>
+                <DxcActionIcon
+                  icon={row.contentIsExpanded ? "arrow_drop_down" : "arrow_right"}
+                  title="icon"
+                  onClick={() => {
+                    row.contentIsExpanded = !row.contentIsExpanded;
+                    if (row.contentIsExpanded) {
+                      const rowIndex = rowsToRender.findIndex((rowToRender) => row === rowToRender);
+                      setRowsToRender((rows) => {
+                        const newRows = [...rows];
+                        addRow(newRows, rowIndex + 1, {
+                          isExpandedChildContent: row.contentIsExpanded,
+                          uniqueRowId: rowKeyGetter(row, uniqueRowId) + "_expanded",
+                          expandedChildContent: row.expandedContent,
+                          triggerRowKey: rowKeyGetter(row, uniqueRowId),
+                          expandedContentHeight: row.expandedContentHeight,
+                        });
+                        return newRows;
+                      });
+                    } else {
+                      const rowIndex = rowsToRender.findIndex((rowToRender) => row === rowToRender);
+                      setRowsToRender((rows) => {
+                        const newRows = [...rows];
+                        deleteRow(newRows, rowIndex + 1);
+                        return newRows;
+                      });
+                    }
+                  }}
+                />
+              </ActionContainer>
+            );
+          },
+        },
+        ...expectedColumns,
+      ];
+    }
+    if (!expandable && rows.some((row) => Array.isArray(row.childRows) && row.childRows.length > 0) && uniqueRowId) {
+      // only the first column will be clickable and will expand the rows
+      const firstColumnKey = expectedColumns[0].key;
+      expectedColumns[0] = {
+        ...expectedColumns[0],
+        renderCell({ row }) {
+          if (row.childRows?.length) {
+            return (
+              <HierarchyContainer level={row.rowLevel || 0}>
+                {renderHierarchyTrigger(rowsToRender, row, uniqueRowId, firstColumnKey, (rows) =>
+                  setRowsToRender(rows)
+                )}
+              </HierarchyContainer>
+            );
+          }
+          return (
+            <HierarchyContainer level={row.rowLevel || 0} className="ellipsis-cell">
+              {row[firstColumnKey]}
+            </HierarchyContainer>
+          );
+        },
+      };
```

Comment:
I don't there is much benefit here, if we want to reduce more, we would need to use styled components inside utils and i personally prefer to avoid that.

Reply:
Okay, it's totally up to you, more personal preference that anything, there is always the option of having those column modifications functions inside the `Datagrid.tsx` file itself just for you to consider. 

Feel free to mark this conversation as resolved.